### PR TITLE
Generate citizen respondent aos invitation

### PR DESCRIPTION
**Pull request by mapingo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-815

### Change description ###
Generate citizen respondent AoS invitation PDF and add to generated documents, if respondent is not represented

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerIssueApplicationTest.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ void shouldGenerateAosPackAndPostAosPackAndSendEmailsWhenRespondentIsRepresented
         );
 
         when(serviceTokenGenerator.generate()).thenReturn(TEST_SERVICE_AUTH_TOKEN);
-        when(documentIdProvider.documentId()).thenReturn("Divorce application").thenReturn("Respondent Invitation");
+        when(documentIdProvider.documentId()).thenReturn("Respondent Invitation").thenReturn("Divorce application");
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerIssueApplicationTest.java:160]: Invitation to invitation ?

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> caseDetails) {
+
+        final Long caseId = caseDetails.getId();
+        final CaseData caseData = caseDetails.getData();
+        final LocalDate createdDate = caseDetails.getCreatedDate().toLocalDate();
+
+        if (!caseData.getApplicant2().isRepresented()) {
+
+            log.info("Generating access code to allow the respondent to access the joint application");
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java:47]: not sure how do we know if it is a joint application ?

Can we not have sole application where respondent is not solicitor represented ?

File: src/integrationTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerIssueApplicationTest.java (hunk 1 above)
Comment (mapingo) [reply src/integrationTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerIssueApplicationTest.java:160]: These were the wrong way round, the Respondent Invitation is generated first

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java (hunk 2 above)
Comment (mapingo) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/task/GenerateCitizenRespondentAosInvitation.java:47]: I will change the message, I forgot to change the log message after copy and paste.

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**